OpenDreamRuntime/Procs/DMProc.cs

```diff
@@ -189,6 +192,9 @@ sealed class DMProcState : ProcState {
         public int ArgumentCount;
         public string? CurrentSource;
         public int CurrentLine;
+        public int CatchPosition = NoTryCatch;
+        public int CatchVarIndex = NoTryCatch;
```

Comment:
These should be reset in either `Dispose()` or `Initialize()`

Proc states can be saved for later reuse, so it's good to not keep values between uses.